File: Programmers/[17687] n진수 게임/SUbbb/17687.java

class Solution {
    public String solution(int n, int t, int m, int p) {
        String[] answer = new String[t];
        int num = 0;
        int idx = 0;
        int order = 1;

        while (idx != t) {
            String strNum = Integer.toString(num, n);

            for (int i = 0; i < strNum.length() && idx != t; i++) {
                if (order == p) answer[idx++] = String.valueOf(strNum.charAt(i)).toUpperCase();
                order++;
                if (order == m + 1) order = 1;
            }
            num++;
        }
        
        return String.join("", answer);
    }
}

File: Programmers/[17687] n진수 게임/KingDonggyu/17687.js
function solution(n, t, m, p) {
  const ruleNumbers = [];
  let number = 0;

  while (t * m > ruleNumbers.length) {
    ruleNumbers.push(...number.toString(n).split(''));
    number++;
  }

  let answer = '';
  
  while(t--) {
    answer += ruleNumbers[p - 1];
    p += m;
  }

  return answer.toUpperCase();
}

console.log(solution(16, 16, 2, 2));